api/v1alpha1/common_validation.go

```diff
@@ -34,37 +34,67 @@ const (
 	ValidateValueParseError = "parse value field error:%s"
 )
 
-// ValidateScheduler validates the scheduler and duration
-func ValidateScheduler(duration *string, scheduler *SchedulerSpec, spec *field.Path) field.ErrorList {
+// ValidateScheduler validates the InnerSchedulerObject
+func ValidateScheduler(schedulerObject InnerSchedulerObject, spec *field.Path) field.ErrorList {
+
 	allErrs := field.ErrorList{}
+
+	nextStart := schedulerObject.GetNextStart()
+	nextRecover := schedulerObject.GetNextRecover()
+	if nextStart.After(nextRecover) {
+		nextStartField := spec.Child("nextStart")
+		allErrs = append(allErrs, field.Invalid(nextStartField, nextStart,
+			fmt.Sprintf("the nextStart: \"%s\" can not after the nextRecover: \"%s\"", nextStart, nextRecover)))
+	}
+
 	schedulerField := spec.Child("scheduler")
 	durationField := spec.Child("duration")
+	duration, err := schedulerObject.GetDuration()
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(durationField, nil,
+			fmt.Sprintf("parse duration field error:%s", err)))
+	}
 
-	if duration != nil && scheduler != nil {
-		_, err := cron.ParseStandard(scheduler.Cron)
-		if err != nil {
-			allErrs = append(allErrs, field.Invalid(schedulerField.Child("cron"),
-				scheduler.Cron,
-				fmt.Sprintf("parse cron field error:%s", err)))
-		}
+	scheduler := schedulerObject.GetScheduler()
 
-		_, err = time.ParseDuration(*duration)
-		if err != nil {
-			allErrs = append(allErrs, field.Invalid(durationField,
-				*duration,
-				fmt.Sprintf("parse duration field error:%s", err)))
+	if duration != nil && scheduler != nil {
+		errs := validateSchedulerParams(duration, durationField, scheduler, schedulerField)
+		if len(errs) != 0 {
+			allErrs = append(allErrs, errs...)
 		}
+	} else if (duration == nil && scheduler != nil) || (duration != nil && scheduler == nil) {
```

Comment:
I think `duration==nil && scheduler != nil` is ok, sometimes we just want to start to chaos experiment at a specific time and last to the end time.

Reply:
I am not sure about that.
But I find the controllers except podchaos all need  `scheduler == nil && duration == nil` OR `scheduler != nil && duration != nil`

https://github.com/pingcap/chaos-mesh/blob/8e769a704563d2becd5bd9269961783bca912952/controllers/iochaos/types.go#L47-L51


https://github.com/pingcap/chaos-mesh/blob/8e769a704563d2becd5bd9269961783bca912952/controllers/stresschaos/types.go#L59-L63

https://github.com/pingcap/chaos-mesh/blob/8e769a704563d2becd5bd9269961783bca912952/controllers/kernelchaos/types.go#L66-L70

https://github.com/pingcap/chaos-mesh/blob/8e769a704563d2becd5bd9269961783bca912952/controllers/timechaos/types.go#L58-L62


But podschaos has a different logic

https://github.com/pingcap/chaos-mesh/blob/8e769a704563d2becd5bd9269961783bca912952/controllers/podchaos/types.go#L48-L52



I think the main reason for this difference is that `PodKillAction` and `ContainerKillAction`  have different behaviors. We have addressed this difference in webhook of podschaos.

Reply:
This can be simplified to `else if !(duration == nil && scheduler == nil)`